### PR TITLE
IOTDB-734 Add Support for NaN in Double / Floats in SQL Syntax.

### PR DESCRIPTION
**Pull request by JulianFeinauer**

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/utils/TypeInferenceUtils.java [hunk 1]
```diff
@@ -65,6 +64,9 @@ public static TSDataType getPredictedDataType(Object value, boolean inferType) {
         } else {
           return floatingStringInferType;
         }
+        // "NaN" is returned if the NaN Literal is given in Parser
+      } else if ("NaN".equals(strValue)) {
+        return TSDataType.DOUBLE;
```
[the remark this answers is not in the document]
Comment (JulianFeinauer) [reply server/src/main/java/org/apache/iotdb/db/utils/TypeInferenceUtils.java:69]: Good Hint. Personally I would see Nan as a Double Literal and thus probably use the check on floating_string_infer_type. WDYT?

[the remark this answers is not in the document]
Comment (qiaojialin) [reply server/src/main/java/org/apache/iotdb/db/utils/TypeInferenceUtils.java:69]: I prefer to add a configuration nan_string_infer_type=DOUBLE  by default and enable users to change this to TEXT.

[the remark this answers is not in the document]
Comment (JulianFeinauer) [reply server/src/main/java/org/apache/iotdb/db/utils/TypeInferenceUtils.java:69]: Yes, that makes sense to me. I will add that option (if I find out how this works...:D).

[the remark this answers is not in the document]
Comment (qiaojialin) [reply server/src/main/java/org/apache/iotdb/db/utils/TypeInferenceUtils.java:69]: You could add nan_string_infer_type=DOUBLE in server/src/assembly/resources/conf/iotdb-engine.properties， Then add itmes in IoTDBDescriptor and IoTDBConfig class. 
After adding this, the docs should be updated in docs/UserGuide/Server/Config Manual.md.

I will add some system design doc for how to do these things.

[the remark this answers is not in the document]
Comment (JulianFeinauer) [reply server/src/main/java/org/apache/iotdb/db/utils/TypeInferenceUtils.java]: Thanks for the advice, I have done it. And also added notes for the other inference config properties as they were missing so far.

**Pull request merged**